astropy/stats/tests/test_sigma_clipping.py

```diff
@@ -174,6 +180,89 @@ def test_sigma_clipped_stats_masked_col():
     sigma_clipped_stats(col)
 
 
+@pytest.mark.parametrize("units", [False, True])
+@pytest.mark.parametrize("sigma", np.linspace(2, 5, 10))
+def test_sigmaclippedstats_stats(units, sigma):
+    """Test SigmaClippedStats class."""
+    data = np.ones((101, 205))
+    rng = np.random.default_rng(0)
+    idx = rng.integers(0, 100, 10)
+    data[idx, idx] = 1000
+
+    if units:
+        unit = u.m
+        data <<= unit
+    else:
+        unit = 1
+
+    stats = SigmaClippedStats(data, sigma=sigma)
+    assert stats.min() == 1.0 * unit
+    assert stats.max() == 1.0 * unit
+    assert stats.sum() == np.sum(data) - (1000 * 10) * unit
+    assert stats.mean() == 1.0 * unit
+    assert stats.median() == 1.0 * unit
+    assert stats.mode() == 1.0 * unit
+    assert stats.mode(median_factor=3.5, mean_factor=2.5) == 1.0 * unit
+    assert stats.std() == 0.0 * unit
+    assert stats.var() == 0.0 * unit**2
+    assert stats.mad_std() == 0.0 * unit
+    assert stats.biweight_location() == 1.0 * unit
+    assert stats.biweight_scale() == 0.0 * unit
+
+    # test nanvar with Angle
+    data = Angle([10, 20, 30, 40], "deg")
+    stats = SigmaClippedStats(data, sigma=5)
+    assert isinstance(stats.mean(), u.Quantity)
+    assert stats.mean() == 25 * u.deg
+    assert stats.var() == 125 * u.deg**2
```

Comment:
Maybe add here
```
assert not isinstance(stats.var(), Angle)
assert isinstance(stats.var(), Quantity)
```

Reply:
@mhvk The tests passed locally for me, but are failing in CI because of the `Angle` input.  Does `_result_as_quantity` require a certain numpy version?
See https://github.com/astropy/astropy/actions/runs/11481861464/job/31953565303?pr=17221

Reply:
I though I had this `assert isinstance(stats.var(), Quantity)` in the tests!  (`stats.mean()` was actually a typo).  I've added these, but I'll wait to push until I can figure out why tests are failing.

Reply:
The error is:
```
E   astropy.units.errors.UnitTypeError: Cannot store output with unit 'deg2' from multiply function in <class 'astropy.coordinates.angles.core.Angle'> instance.  Use <class 'astropy.units.quantity.Quantity'> instance instead.

astropy/units/quantity_helper/converters.py:343: UnitTypeError
```

Reply:
This test has uncovered an error in `sigma_clip`.  The error occurs in `sigma_clip` when `bottleneck` is uninstalled.  Here's a MWE.  @mhvk Any ideas?

```python
from astropy.stats import sigma_clip
from astropy.coordinates import Angle
a = Angle([10, 20, 30, 40], 'deg')
sigma_clip(a)
```

```
UnitTypeError: Cannot store output with unit 'deg2' from multiply function in <class 'astropy.coordinates.angles.core.Angle'> instance.  Use <class 'astropy.units.quantity.Quantity'> instance instead.
```

Reply:
The surprising root cause is in `np.nanstd`.  Should we disable `Angle` input to sigma clipping functions?

```python
import bottleneck as bn
import numpy as np
from astropy.coordinates import Angle
from astropy.stats.nanfunctions import nanstd

a = Angle([10, 20, 30, 40], 'deg')

np.std(a)
<Angle 11.18033989 deg>

np.nanstd(a)
UnitTypeError

bn.nanstd(a)
11.180339887498949

nanstd(a)
<Angle 11.18033989 deg>
```

Reply:
OK, that's clearly a `units` bug - will look into it!

Reply:
Now hopefully fixed with a rebase!

Reply:
Rebased.  The tests passed locally now.